Sources/MediaCategories/MediaCategoryViewController.swift

```diff
@@ -17,13 +17,14 @@ protocol MediaCategoryViewControllerDelegate: NSObjectProtocol {
     func needsToUpdateNavigationbarIfNeeded(_ viewController: VLCMediaCategoryViewController)
 }
 
-class VLCMediaCategoryViewController: UICollectionViewController, UICollectionViewDelegateFlowLayout, UISearchResultsUpdating, UISearchControllerDelegate, IndicatorInfoProvider {
+class VLCMediaCategoryViewController: UICollectionViewController, UICollectionViewDelegateFlowLayout, UISearchBarDelegate, IndicatorInfoProvider {
 
     var model: MediaLibraryBaseModel
-
+    var searchBar = UISearchBar(frame: .zero)
+    private var searchbarConstraint: NSLayoutConstraint?
```

Comment:
What would you think of using a closure to initialize this lazily?

Indeed, this can be confusing if we don't need to actually use the optionality.

On the nit-picking side, I realize that it isn't your code but could we put the search properties grouped together?
Additionally, capitalization missing on `searchBarConstraint`.

Reply:
I believe it is more confusing to make it lazily intialized because you rip it out of context with the other searchbar constraints. 
I can group the properties together yes